lib/activerecord-multi-tenant/model_extensions.rb

```diff
@@ -6,7 +6,15 @@ def multi_tenant(tenant_name, options = {})
       if to_s.underscore.to_sym == tenant_name
         unless MultiTenant.with_write_only_mode_enabled?
           # This is the tenant model itself. Workaround for https://github.com/citusdata/citus/issues/687
-          before_create -> { self.id ||= self.class.connection.select_value("SELECT nextval('" + [self.class.table_name, self.class.primary_key, 'seq'].join('_') + "'::regclass)") }
+          before_create lambda {
+            self.id ||= if self.class.columns_hash['id'].type == :uuid
```

Comment:
Should this be `columns_hash[self.class.primary_key]` instead?

Reply:
Hi @serprex , for shure! 
I updated with your suggestion.